boilerplate/openshift/golang-osd-operator/standard.mk

```diff
@@ -74,17 +78,17 @@ docker-build: isclean
 	${CONTAINER_ENGINE} tag $(OPERATOR_IMAGE_URI) $(OPERATOR_IMAGE_URI_LATEST)
 
 .PHONY: docker-push
-docker-push: docker-build
+docker-push:
 	${CONTAINER_ENGINE} push $(OPERATOR_IMAGE_URI)
 	${CONTAINER_ENGINE} push $(OPERATOR_IMAGE_URI_LATEST)
 
 .PHONY: push
 push: docker-push
 
 .PHONY: go-check
-go-check: # golanci-lint and other static analysis
+go-check: ## Golang linting and other static analysis
 	${CONVENTION_DIR}/ensure.sh golangci-lint
-	GOLANGCI_LINT_CACHE=${GOLANGCI_LINT_CACHE} golangci-lint run -c ${GOLANGCI_LINT_CONFIG} ./...
+	GOLANGCI_LINT_CACHE=${GOLANGCI_LINT_CACHE} golangci-lint run -c ${CONVENTION_DIR}/golangci.yml ./...
```

Comment:
This is going to override our own config that it was possible to set before... I'll fix it in a subsequent PR

Reply:
We're trying to avoid allowing too much customization by consumers, as it sort of defeats the purpose. What did you have in mind?

Reply:
I would not assume every consumer of boilerplate will have the same ideas about linting and static code analysis, e.g

* https://github.com/openshift/aws-account-operator/blob/master/.golangci.yml
* https://github.com/openshift/hive/blob/master/golangci.yml
* https://github.com/openshift/telemeter/blob/master/.golangci.yml
* https://github.com/app-sre/deployment-validation-operator/blob/master/.golangci.yml

Reply:
Right now they don't, I agree. Converging on that is one of the primary pushes of the boilerplate effort. Common linting rules across repositories has $benefits :)

Reply:
This is what I had in mind for boilerplate: https://github.com/app-sre/deployment-validation-operator/pull/39